src/client-mixins/request-handler.helper.ts

```diff
@@ -155,25 +155,43 @@ export class RequestHandlerHelper<T> {
     return res;
   }
 
-  protected getParsedResponse(res: IncomingMessage) {
-    let data: any = this.responseData;
-
+  protected detectResponseType(res: IncomingMessage): TResponseParseMode {
     // Auto parse if server responds with JSON body
-    if (data.length && res.headers['content-type']?.includes('application/json')) {
-      data = JSON.parse(data);
+    if (res.headers['content-type']?.includes('application/json') || res.headers['content-type']?.includes('application/problem+json')) {
+      return 'json';
     }
     // f-e oauth token endpoints
     else if (this.isFormEncodedEndpoint()) {
+      return 'url';
+    }
+
+    return 'text';
+  }
+
+  protected getParsedResponse(res: IncomingMessage) {
+    const data = this.responseData;
+    const mode = this.requestData.forceParseMode || this.detectResponseType(res);
+
+    if (mode === 'buffer') {
```

Comment:
[totally optional/nitpicking]: what about a switch-case?

Reply:
I don't like switches, especially when the condition block contains more than 1 instruction (case for json and url). That's why, but I can convert this code to `switch` if you want to.

Reply:
no need to, thx for the explanation :)